api/api.go

```diff
@@ -495,8 +497,29 @@ func (b *BlockChainAPI) EstimateGas(
 	blockNumberOrHash *rpc.BlockNumberOrHash,
 	overrides *StateOverride,
 ) (hexutil.Uint64, error) {
-	// todo we return max gas limit until we have gas estimation in place
-	return hexutil.Uint64(30_000_000), nil
+	var data []byte
+	if args.Data != nil {
+		data = *args.Data
+	} else if args.Input != nil {
+		data = *args.Input
+	}
```

Comment:
what if neither, have another else with error

Reply:
Actually, the TX can also be a value transfer, so it is legit to not provide `input` or `data` fields.